PocketTutor-Team/pockettutor-app
# feature/update main

# Implement navigations between screens and Refactor directory architecture and filenames

### PR description
This PR uses the already implemented NavigationAction class to navigate between the already implemented screens from the Sign-In and the Profile creation (both for Tutor and Student) to the overview screen.
It also rename some directories name and filenames ot have a clean arhcitecture with explicit name, that are coherent between them.

### Summary
- Update MainActivity to navigate between already implemented screens (AUTH, HOME, CREATE_PROFILE, CREATE_TUTOR_PROFILE. CREATE_TUTOR_SCHEDULE).
- Add a googleUid field in the Profile dataclass to be able to link the google account of the users with its profile.
- Add navigation and repository update for the CreateTutorProfileScreen and CreateTutorScheduleScreen when the user click on the validation button.

#### Pipeline of the navigation
First we have the Sign-In screen (AUTH). After having log-in with google sign-in, there is two paths: either we already have an profile and we directly go to the overview screen (HOME), or we need to create an account and we go to the create profile screen (CREATE_PROFILE).
In the create profile screen, we will need to enter the field to give our personnal information and then when we click on the validation button, we also have two paths. If we choose to be a student, we go to the overview screen (HOME). If we choose to be a tutor, we go to the create tutor profile screen (CREATE_TUTOR_PROFILE). After validation in this screen, we go to the create schedule for tutor screen (CREATE_TUTOR_SCHEDULE). After validating in this screen we finally go to the overview screen (HOME). 

### How to test
To test this PR, one can run the MainActivity and check that the pipeline describe before works and that the screen corresponds to the screenshots given below.
It could be useful two create to profiles, one which is a Tutor and one which is a Student.

### Screenshots
(In the order given: AUTH Screen (for Sign-In), HOME Screen (for now Greeting screen but will be update later), CREATE_PROFILE Screen, CREATE_TUTOR_PROFILE Screen, CREATE_TUTOR_SCHEDULE Screen
![image](https://github.com/user-attachments/assets/38cc0f25-7915-4f1a-b128-e201df5cc07a) ![image](https://github.com/user-attachments/assets/542bd7ad-d30a-4e23-9000-a645c408464c) ![image](https://github.com/user-attachments/assets/ecf7a4b8-3367-4873-9dbb-e419282ac52b) ![image](https://github.com/user-attachments/assets/0a0b6094-4771-420a-8ce7-02e6cc85060f) ![image](https://github.com/user-attachments/assets/ccb13dd9-d547-4091-a6cf-4f326c8cf92a)
